cmd/faucet/faucet.go

```diff
@@ -629,7 +725,15 @@ func (f *faucet) apiHandler(w http.ResponseWriter, r *http.Request) {
 			amount = new(big.Int).Div(amount, new(big.Int).Exp(big.NewInt(2), big.NewInt(int64(msg.Tier)), nil))
 
 			tx := types.NewTransaction(f.nonce+uint64(len(f.reqs)), address, amount, 21000, f.price, nil)
-			signed, err := f.keystore.SignTx(f.account, tx, f.config.GetChainID())
+
+			// FIXME(meowsbits): Getting the chain id more than once is redundant and can be optimized.
```

Comment:
Probably there are at least a few ways of handling it. One that comes first to mind is to add a field on `faucet`, eg. `faucet.chainID` since we know that once it's set to a nonzero value it won't change. But IMO this is a nice-to-have optimization since the request will only be made for each faucet transaction... which isn't more than can be expected for a "standard" transaction-making program (at least it's not getting called at every block or anything like that). Because of this I'm OK pushing to a later optimization if need/desire calls for it.